app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+                toolbarShown = toolbarState.toolbarShown,
+                onToolbarShownUpdate = { newValue ->
+                    toolbarState = toolbarStateFromBoolean(newValue)
+                },
+                plant = plant,
+                isPlanted = isPlanted,
+                callbacks = callbacks,
+                transitionState = transitionState
+            )
+            PlantHeader(toolbarState.toolbarShown, plant.name, callbacks, transitionState)
+        }
+    }
+}
+
+@Composable
+private fun PlantDetailsContent(
+    scrollerPosition: ScrollerPosition,
+    toolbarShown: Boolean,
+    onToolbarShownUpdate: (Boolean) -> Unit,
+    plant: Plant,
+    isPlanted: Boolean,
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    VerticalScroller(scrollerPosition) {
+        // The header transition happens given the _original_ position of the name on the screen
+        var namePosition by state { Float.MAX_VALUE }
+        // Whenever the name position or the scroller position changes,
+        // check if the toolbar should be shown or not
+        onCommit(namePosition, scrollerPosition.value) {
+            onToolbarShownUpdate(
+                scrollerPosition.value > (namePosition + HeaderTransitionOffset)
+            )
+        }
+
+        Hide(toolbarShown) { hideModifier ->
+            PlantImageHeader(
+                scrollerPosition, plant.imageUrl, callbacks.onFabClicked, isPlanted, hideModifier,
+                Modifier.drawLayer(alpha = transitionState[contentAlphaKey], clip = false)
+            )
+        }
+        PlantInformation(
+            name = plant.name,
+            wateringInterval = plant.wateringInterval,
+            description = plant.description,
+            onNamePositioned = {
+                if (namePosition == Float.MAX_VALUE) {
+                    namePosition = it.globalPosition.y
+                }
+            },
+            toolbarShown = toolbarShown
+        )
+    }
+}
+
+@Composable
+private fun PlantHeader(
+    toolbarShown: Boolean,
+    plantName: String,
+    callbacks: PlantDetailsCallbacks,
+    transitionState: TransitionState
+) {
+    if (toolbarShown) {
+        PlantDetailsToolbar(
+            plantName = plantName,
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[toolbarAlphaKey])
+        )
+    } else {
+        PlantHeaderActions(
+            onBackClicked = callbacks.onBackClicked,
+            onShareClicked = callbacks.onShareClicked,
+            modifier = Modifier.drawOpacity(transitionState[contentAlphaKey])
+        )
+    }
+}
+
+@Composable
+private fun PlantDetailsToolbar(
+    plantName: String,
+    onBackClicked: () -> Unit,
+    onShareClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier) {
+        val spacerHeight = with(DensityAmbient.current) {
```

Comment:
Using `Modifier.systemBarsPadding(top = true)` on `TopAppBar` seems cleaner here.

Reply:
I don't need padding here, I need the height because I want to fill this space with the `surface` color

Reply:
That feels like a smell that you're missing an outer container with `surface` color that this padding reveals?

Reply:
That'd be true in a normal case, but in this one, the plant image would be displayed instead of a surface. Maybe I should add a comment to clarify this

Reply:
Done as chatted offline! Thanks